hv.c

```diff
@@ -480,7 +480,7 @@ Perl_hv_common(pTHX_ HV *hv, SV *keysv, const char *key, STRLEN klen,
 
     if (!hv)
         return NULL;
-    if (SvTYPE(hv) == (svtype)SVTYPEMASK)
+    if ((U8)SvTYPE(hv) == (U8)SVTYPEMASK)
```

Comment:
These should all use `SvIS_FREED(sv)`.

(I started typing "Maybe we should have a macro...".)

Reply:
Hmm.  I didnt know we had that one. But the code would be a little different:

```
sv.h:#define SvIS_FREED(sv)     UNLIKELY(((sv)->sv_flags == SVTYPEMASK))
sv.h:#define SvTYPE(sv) ((svtype)((sv)->sv_flags & SVTYPEMASK))
```

Are you confident that would be safe?